Enoch/ac 34 people api doc timezone

ExtraGravity opened the pull request.

ExtraGravity committed:
ac-34 update people api to include a timezone

ExtraGravity committed:
ac-34 made update attributes optional

ExtraGravity committed:
Add milliseconds to last modified spec

Last modified is being used by client to determine whether to sync or not.
So having a smaller granularity would be helpful.

beaesguerra commented on src/routers/people.ts:
```diff
@@ -49,10 +49,10 @@ router.get('/', async (req, res) => {
- *              specifies what kind of people are given.
- * @apiName GetPeopleCategory
+ * @apiDescription Returns all people of this category and all their
+ * 				   associated attributes and values.
```
weird tabbings?

beaesguerra commented on src/routers/people.ts:
```diff
@@ -62,15 +62,15 @@ router.get('/', async (req, res) => {
   [
     {
       "personUuid": "1e167b81-d816-497b-8c0c-36f4d6b2fd33",
+      "lastModified": "2018-01-23 04:05:06Z.123"
```
Z is after ms?

beaesguerra commented on src/routers/people.ts:
```diff
@@ -62,15 +62,15 @@ router.get('/', async (req, res) => {
       "personUuid": "4b3f23a3-04c4-468f-bdf5-f189a34d9f69",
       "firstName": "Zachariah",
-      "lastModified": "2018-01-23 04:05:06"
+      "lastModified": "2018-01-23 04:05:06Z.123"
```
z is after ms

beaesguerra commented on src/routers/people.ts:
```diff
@@ -88,7 +88,6 @@ router.get('/:category', async (req, res) => {
  * @apiDescription Creates a new person in the specified category.
  *              Returns the UUID created for that person.
  *              Requires all associated attributes given in /peopleCategories
- * @apiName CreatePerson
```
why was this removed?

ExtraGravity replied:
It's on the lines above, duplicated.

beaesguerra commented on src/routers/people.ts:
```diff
@@ -121,7 +119,7 @@ router.post('/:category', async (req, res) => {
    {
      "personUuid": "1e167b81-d816-497b-8c0c-36f4d6b2fd33",
-     "lastModified": "2018-01-23 04:05:06"
+     "lastModified": "2018-01-23 04:05:06Z.123"
```
z is after ms

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+/**
+ * @api {post} /transactions/products/<type>
+ * @apiDescription Returns all product transactions of a specified type.
+ *              Required fields are:
```
use apiparam please!

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+ *
+ * @apiName CreateNewProductTransaction
+ *
+ * @apiError (400) The required fields are invalid or missing, ...
```
need the "badrequest" part?

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+ * @api {put} /transactions/products/<type>/<uuid>
+ * @apiDescription Updates a product transaction.
+ *              Available fields are:
+ *                - datetime
```
use apiparam please!

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+ *
+ * @apiName UpdateProductTransaction
+ *
+ * @apiError (400) The fields are invalid, ...
```
need "badrequest"

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+ *
+ * @apiError (400) The fields are invalid, ...
+ * @apiError (401) Unauthorized - Must be admin
+ * @apiError (404) The product transaction has not been found.
```
need "notfound"

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -8,14 +8,96 @@ import { StatusCode } from '@/models/statusCodes';
+ * @apiError (401) Unauthorized - Must be admin
+ * @apiError (404) The product transaction has not been found.
+ *
+ * @apiSuccess (200) {String} Successfully updated <type> transaction
```
need "success"

ExtraGravity committed:
Merge branch 'master' into enoch/ac-34-people-api-doc-timezone

ExtraGravity committed:
ac-34 Respond to pr #46 comments

Merged with master. Fixed times where the Z was not after the milliseconds
but in the middle of the time. Also fixed url parameters not documented
as an apiparam.
